### PR TITLE
Specify Release configuration for test build, fix some optimization issues

### PR DESCRIPTION
**Pull request by ikesyo**
Depends on #1902. To detect issues caused by the compiler optimization.

Now unfortunately, [this test](https://github.com/ReactiveCocoa/ReactiveCocoa/blob/091d09960df11c3538b9e59f732d1e3ad8bb2e97/ReactiveCocoaTests/Objective-C/RACKVOWrapperSpec.m#L623) is failing :cry:

### CODE REVIEW

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa iOS.xcscheme [hunk 1]
```diff
@@ -40,7 +40,7 @@
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       shouldUseLaunchSchemeArgsEnv = "YES"
-      buildConfiguration = "Debug">
+      buildConfiguration = "Release">
```
Comment (NachoSoto) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa iOS.xcscheme]: :+1:

**Comment by jspahrsummers** [comment]
Thanks @ikesyo! This looks great. :sparkles:

That one test is failing because it's assigning directly into a `__weak` variable, so it will disappear almost immediately with optimization. I think we could either make it strong, or else create a temporarily strong reference to it just within the `@autoreleasepool`.

**Comment by ikesyo** [comment]
@jspahrsummers I fixed that with the latter way. :boom:

**Comment by jspahrsummers** [comment]
Looks good! :shipit:

**Pull request merged**